lib/src/executable.dart

```diff
@@ -37,29 +47,30 @@ final _relativeDevDartPath = p.relative(
 
 Future<void> run(List<String> args) async {
   attachLoggerToStdio(args);
-  final configExists = File(_configPath).existsSync();
-  final oldDevDartExists = File(_oldDevDartPath).existsSync();
 
-  if (!configExists) {
+  if (!_config.existsSync()) {
     log.fine('No custom `tool/dart_dev/config.dart` file found; '
         'using default config.');
   }
-  if (oldDevDartExists) {
+  if (_oldDevDart.existsSync()) {
     log.warning(yellow.wrap(
         'dart_dev v3 now expects configuration to be at `$_configPath`,\n'
         'but `$_oldDevDartPath` still exists. View the guide to see how to upgrade:\n'
         'https://github.com/Workiva/dart_dev/blob/master/doc/v3-upgrade-guide.md'));
   }
 
-  if (args.contains('hackFastFormat') && !oldDevDartExists) {
+  if (args.contains('hackFastFormat') && !_oldDevDart.existsSync()) {
     await handleFastFormat(args);
-
     return;
   }
 
-  generateRunScript();
+  final processArgs = generateRunScript();
   final process = await Process.start(
-      Platform.executable, [_runScriptPath, ...args],
+      processArgs.first,
+      [
+        if (processArgs.length > 1) ...processArgs.sublist(1),
```

Comment:
#nit This check is unnecessary, since for a length of 1 .sublist will return an empty list:
```suggestion
        ...processArgs.sublist(1),
```